bug: incorrectly handle alias type of pointer in struct

## Reproductible codes
```
type Issue403 struct {
	Card   PtrAlias `json:"card"`
}

type PtrAlias *MarshalerImpl

type MarshalerImpl struct{
	A int
}

func (self *MarshalerImpl) MarshalJSON() ([]byte, error) {
	return []byte("1"), nil
}

func TestIssue403(t *testing.T) {
	obj := MarshalerImpl{0}
	messageInfo := &Issue403{
		Card:    &obj,
	}

	jsonData, err := json.Marshal(messageInfo)
	require.NoError(t, err)
	sonicData, err := sonic.Marshal(messageInfo)
	require.NoError(t, err)
	require.Equal(t, jsonData, sonicData)
}
```
## Result
```
--- FAIL: TestIssue403 (0.01s)
    /Users/admin/GOPATH/work/sonic/issue_test/issuex_test.go:51: 
        	Error Trace:	/Users/admin/GOPATH/work/sonic/issue_test/issuex_test.go:51
        	Error:      	Not equal: 
        	            	expected: []byte{0x7b, 0x22, 0x63, 0x61, 0x72, 0x64, 0x22, 0x3a, 0x31, 0x7d}
        	            	actual  : []byte{0x7b, 0x22, 0x63, 0x61, 0x72, 0x64, 0x22, 0x3a, 0x6e, 0x75, 0x6c, 0x6c, 0x7d}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	-([]uint8) (len=10) {
        	            	- 00000000  7b 22 63 61 72 64 22 3a  31 7d                    |{"card":1}|
        	            	+([]uint8) (len=13) {
        	            	+ 00000000  7b 22 63 61 72 64 22 3a  6e 75 6c 6c 7d           |{"card":null}|
        	            	 }
        	Test:       	TestIssue403
```
## Reason
in internal/resolver.go:resolveFields(), sonic incorrectly store field's pointer type instead of final concrete type's pointer type

--- a/internal/resolver/resolver.go
+++ b/internal/resolver/resolver.go
@@ -152,8 +152,8 @@ func resolveFields(vt reflect.Type) []FieldMeta {
         }
 
         /* get the index to the last offset */
-        fvt := fv.typ
         idx := len(path) - 1
+        fvt := path[idx].Type
 
         /* do not dereference into fields */
         if path[idx].Kind == F_deref {

--- a/issue_test/issuex403_test.go
+++ b/issue_test/issuex403_test.go
@@ -1,0 +1,52 @@
+/**
+ * Copyright 2023 ByteDance Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package issue_test
+
+import (
+    `encoding/json`
+    `testing`
+
+    `github.com/bytedance/sonic`
+    `github.com/stretchr/testify/require`
+)
+
+type Issue403 struct {
+    Card   PtrAlias `json:"card"`
+}
+
+type PtrAlias *MarshalerImpl
+
+type MarshalerImpl struct{
+    A int
+}
+
+func (self *MarshalerImpl) MarshalJSON() ([]byte, error) {
+    return []byte("1"), nil
+}
+
+func TestIssue403(t *testing.T) {
+    obj := MarshalerImpl{0}
+    messageInfo := &Issue403{
+        Card:    &obj,
+    }
+
+    jsonData, err := json.Marshal(messageInfo)
+    require.NoError(t, err)
+    sonicData, err := sonic.Marshal(messageInfo)
+    require.NoError(t, err)
+    require.Equal(t, jsonData, sonicData)
+}

